app/src/main/res/layout/activity_new_problem.xml

```diff
@@ -15,14 +15,31 @@
             android:layout_height="wrap_content"
             android:orientation="vertical">
 
-            <include
-                android:id="@+id/photo_gallery_layout"
-                layout="@layout/problem_photo_gallery" />
+            <FrameLayout
+                android:layout_width="match_parent"
+                android:layout_height="272dp">
+
+                <include
+                    android:id="@+id/photo_gallery_layout"
+                    layout="@layout/problem_photo_gallery" />
+
+                <android.support.design.widget.FloatingActionButton
+                    android:id="@+id/report_problem_take_photo"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="end|bottom"
+                    android:layout_marginEnd="@dimen/activity_horizontal_margin"
+                    android:layout_marginRight="@dimen/activity_horizontal_margin"
+                    android:layout_marginBottom="8dp"
+                    android:src="@drawable/ic_photo_camera_white" />
+            </FrameLayout>
 
             <GridLayout
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
-                android:layout_margin="28dp"
+                android:layout_marginRight="28dp"
```

Comment:
Margin values doesn't follow baseline grid rules: https://material.google.com/layout/metrics-keylines.html#
Could it be` 32dp` or `24dp`?